gallery/plot_transforms.py

```diff
@@ -178,3 +178,25 @@ def _plot(img, title, **kwargs):
 # performs gaussianblur transform on an image.
 gaus_blur_img = T.GaussianBlur(kernel_size=(5, 9), sigma=(0.4, 3.0))(orig_img)
 plot(gaus_blur_img, "Gaussian Blurred Image")
+
+
+####################################
+# AutoAugment
+# -----------
+# The :class:`~torchvision.transforms.AutoAugment` transform
+# AutoAugments data based on available AutoAugmentation Policies.
+# Use :class:`torchvision.transforms.AutoAugmentPolicy` to create policies.
+# Available  policies are IMAGENET, CIFAR10 and SVHN.
```

Comment:
Note that unless you fix the seed, every time you run this you will get a different picture because AutoAugment has random parts.

Reply:
This happens for other Random transforms too such as RandomResizedCrop, RandomPerspective etc, Should add random seed or a note saying that the outputs might not be same?

Reply:
As suggested in https://github.com/pytorch/vision/issues/3688#issuecomment-823045449 i think it's a good thing not to set the seed. We can add the same note as for the other transforms wrt reproducibility. Or we can also group them in the same "Random transforms " section?

> I didn't set it for the ColorJitter, the idea being that if a user downloads the notebook they will see that a different transformation is performed each time the cell is executed, which is good because it illustrates that these transforms are random at each call.
We should try to set the parameters of the transforms such that it's always obvious that something happened, i.e. that the 2 images are clearly not the same.

Reply:
I think we should be grouping transforms. We can create such sections fmassa too had suggested in [comment](https://github.com/pytorch/vision/pull/3652#pullrequestreview-638851660).

Let's think of grouping and group these

Reply:
I'm not convinced  that a random behaviour is good. I think the intention of this gallery is to showcase the visual result of the transformation to the user which can be problematic for cases such as the `RandomHorizontalFlip` which might not update the original image. As a reader of the documentation I would expect to see the image flipped even if this is a random transform. I think the gallery should visually show the effect to the users. Showing the original image can be confusing.

Reply:
> I think the intention of this gallery is to showcase the visual result of the transformation to the user

more specifically the intention of the gallery is to showcase the behaviour of the **transforms**. For the transforms that are applied randomly, it's good that a few of them showcase that behaviour.

I agree this can be slightly surprising though.

Note that we're careful in the example to warn the user about having potentially the same image (https://pytorch.org/vision/master/auto_examples/plot_transforms.html#randomhorizontalflip):

![image](https://user-images.githubusercontent.com/1190450/116049159-e34c3300-a66d-11eb-83e8-b5586292c867.png)


I think that if we bundle the randomly-applied transforms together, and e.g. sample the same transform a bunch of time to illustrate the randomness, we can deliver a simple message and avoid the confusion.

I'm happy to work on that on a follow-up PR. We'll need all transforms to be included in the example first, so for now I'd suggest to not include the seed and keep on doing as we were doing.

Reply:
> I agree this can be slightly surprising though.

Think of it as if you where writing a blogpost to explain the effects of the Transforms. Would you provide for any of them a single example that had no effect?

I think the documentation avoid such surprises and provide clear visual examples of the transforms. Note that there are some that are not self-explanatory to a beginner. Take for instance `RandomSolarize`; controlling the example to ensure that it will demonstrate the effect can help those users understand the nature of the transform. 

> Note that we're careful in the example to warn

The warning is useful. Have you considered using it the other way? For example, you can force the transform to produce the random effect (via seeding or other method) and then remind them that this transform may or may not apply the effect with probability `p`. 

> I'm happy to work on that on a follow-up PR.

I'm OK if you prefer to tackle this on a follow up PR, as long as it does not get buried in the TODO list. Personally I would set the seed for any effect such as `RandomHorizontalFlip`, `RandomVerticalFlip`, `RandomGrayscale`, `RandomInvert`, `RandomPosterize`, `RandomSolarize` and `RandomEqualize` that are in risk of not showing any effect and seek for an alternative solution later.